### DISCUSSION TITLE
Validate a subtree of a form

### DISCUSSION THREAD

**Discussion by Londeren** [airjp73/rvf#340] [Q&A]
Hi!
I have my form separated into several tabs (no lazy loading here). We can also imagine a wizard here, where it's needed to validate step by step, or tab by tab in my case. After validation, it's needed to get all the values from all the steps and save the whole form to the server.

It's necessary to disable the button on the last step of the wizard if the fields of any step have some errors. I've found [the similar question about a wizard](https://github.com/airjp73/remix-validated-form/issues/306), however, in my case it's not necessary to prevent the user from switching between tabs. And it's necessary to save all the data using a single backend call.

I thought about having separate forms for each tab, but instead of submitting them, getting the data from all of them and do some custom action with all the data.
 
How could I do it using remix-validated-form? I use yup as a validator lib.

Any help would be much appreciated.

**Comment by QuentinWidlocher** [comment]
Bumping this old post because I had the same problem. 

Given a form shaped like this: 

```ts
{
	step1: {
		field1: string,
		field2: number | undefined,
	},
	step2: {
		field1: string,
		field2: boolean,
	}
}
```

I would have loved to do `form.field('step1').dirty()` or `form.field('step1').formState.isValid` for example to know if all field under `step1` are valid (or dirty, or touched).

~~The only way I found was a bit convoluted~~

~~const step1Fields = useFormScope(form.scope("step1"));~~
~~console.log({ ...step1Fields.formState.isDirty });~~
(This returned the state of the form and not this "subform"/"node")

The only way to get the state of the "node" is to know all the "leaf" field names and aggregate their states.
It's not even possible to iterate over each "leaf" fields.

Also please note that the form here is simplified as I would like to be able to do this for a deeply nested object, not just one depth.

Is there a way to do that I missed ? If not, I think it would make for a great feature 👍
